src/controllers/orderController.js

```diff
@@ -57,11 +57,36 @@ const createOrder = async (req, res) => {
   res
     .status(StatusCodes.CREATED)
     .json({ clientSecret: paymentIntent.client_secret, order });
+  /*
+  #swagger.summary = 'Create a new order and process payment'
+  #swagger.description = 'Creates a new order and processes payment using Stripe.'
+  #swagger.tags = ['Orders']
+  #swagger.parameters['body'] = {
+    in: 'body',
+    description: 'Order information including order items, subtotal, tax, and total',
+    required: true,
+    schema: {
+      $ref: '#/definitions/Order' 
```

Comment:
The schema that we expect the client to give us in the body request isn't actually exactly the same as the schema that we return in the response:

![](https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/9094098/580e74a2-b2de-4627-98ad-a94146087fdd)

For example, we don't expect the client to pass in the \_id, paymentIntentId, createdAt, updatedAt, or orderStatus (these are all generated by the backend/controller code), and the user isn't part of the expected request body since the controller will just grab that from the cookie. 

We should create a separate object in the `definitions` block of the `swagger.js` file that describes the shape of the "InputOrder" or "NewOrder" or however you prefer to name it. You can see that we did a similar thing in the `main` branch with `Album` object used for responses and `NewAlbum` object used for the request body: https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/main/swagger.js#L73-L94

Reply:
Akos, I have added new order , commit 913f64ac1afde6be16a90bd348ae1bcb4b4ba7b8

Reply:
You should change line 69 here to refer to: `$ref: '#/definitions/NewOrder'`

Reply:
Thank you, done, commit 85a91c42a5311e2e4542c65fc6bbd4f3fdf1fd59